plugins/data_coverage/src/openjdk/jcov/data/Env.java

```diff
@@ -24,52 +24,90 @@
  */
 package openjdk.jcov.data;
 
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
 import java.lang.reflect.InvocationTargetException;
 import java.nio.file.Path;
 import java.util.Arrays;
 import java.util.Map;
-import java.util.Objects;
-
-import static java.util.stream.Collectors.toMap;
+import java.util.Properties;
+import java.util.Set;
 
 /**
  * Much of the functionality in this plugin is controlled through system properties. This class defines some shortcuts
  * which makes it a bit easier.
  */
 public class Env {
 
-    private static Map<String, String> properties = System.getProperties().entrySet().stream()
-            .collect(toMap(Object::toString, Objects::toString));
+    public static final String PROP_FILE =
+            Env.class.getPackageName().replace('.', '/') +
+            "/instrumentation.properties";
+    /**
+     * Prefix for all system property names which will be passed to the VM running JCov calls.
+     */
+    public static final String JCOV_DATA_ENV_PREFIX = "jcov.data.";
+
+    public static final String JCOV_DATA_SAVER = "jcov.data.saver";
+    private static final String JCOV_OPTION_FOR_DATA_SAVER = "jcov.data-saver";
 
-    public static Map<String, String> properties() {
-        return properties;
+    private static volatile Properties SAVED;
+
+    public static void setSystemProperties(Map<String, String> properties) {
+        properties.forEach((k, v) -> System.setProperty(k, v));
     }
 
-    public static void properties(Map<String, String> properties) {
-        Env.properties = properties;
+    public static void clear(String prefix) {
+        Set<String> keys = System.getProperties().stringPropertyNames();
+        keys.stream().filter(k -> k.startsWith(prefix))
+                .forEach(k -> System.clearProperty(k));
     }
 
     public static String getStringEnv(String property, String defaultValue) {
```

Comment:
defaultValue isn't used and could be omited. 
line 70 of Class Collect  
```
if (!Env.getStringEnv(COVERAGE_IN, "").isEmpty()) {
``` 
could be chaged to 
```
if( Env.getStringEnv(COVERAGE_IN) != null
```
Also isn't clear why you wraps IOException with UncheckedIOException here.

Reply:
in line 115:
} else return defaultValue;

Reply:
Well, I just appeal for the single way for processing all properties If option is missing then it is always NULL otherwise it has a value that should be processed in the place of using. In the program we have many options that are null by default and just one that is empty string.
For this option we have a special branch in the method From my point of view the program should be more restricted If we have m:1 branches of processing something the we should covert it to m+1:0 and remove program logic for this :1 branch.